mllib/src/main/scala/org/apache/spark/ml/feature/ChiSqSelector.scala

```diff
@@ -79,11 +79,12 @@ final class ChiSqSelector(override val uid: String)
 
   @Since("2.0.0")
   override def fit(dataset: Dataset[_]): ChiSqSelectorModel = {
+    val sqlContext = dataset.sqlContext
+    import sqlContext.implicits._
+
     transformSchema(dataset.schema, logging = true)
-    val input = dataset.select($(labelCol), $(featuresCol)).rdd.map {
-      case Row(label: Double, features: Vector) =>
-        LabeledPoint(label, features)
-    }
+    val input = dataset.select(col($(labelCol)).cast(DoubleType).as("label"),
```

Comment:
Sorry, couldn't resist :) I'd change `"label"` to be a symbol `'label`. Not very widely used, but think it deserves its place in the code.